### PR TITLE
[TDV-745] KYC verification queue

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/components/User/KycRequests/KycRequests.Queue.vue [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+    currentDecision () {
+      return this.reviewDecisions
+        .find(item => item.request === this.currentRequest)
+    },
+
+    isCurrentDecisionShown () {
+      return this.currentDecision &&
+        this.currentDecision.action !== DECISION_ACTIONS.skip
+    },
```
Comment (max-nes) [inline src/components/User/KycRequests/KycRequests.Queue.vue]: "skip" should be shown also

File: src/components/User/KycRequests/KycRequests.Queue.vue [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    await this.loadList()
+  },
+
+  destroyed () {
+    window.removeEventListener('beforeunload', this.preventReload)
+  },
+
+  methods: {
+    preventReload (event) {
+      const dialogMessage = 'Changes you made may not be saved.'
```
Comment (max-nes) [inline src/components/User/KycRequests/KycRequests.Queue.vue]: useless, browser ignores it anyway

File: src/components/User/KycRequests/KycRequests.Queue.vue [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    nextRequest () {
+      window.scrollTo({
+        top: 0,
+        left: 0,
+        behavior: 'smooth',
+      })
+      this.currentRequestIndex++
+
+      if (this.isDecisionEdited) {
+        this.isConfirmationShown = true
```
Comment (max-nes) [inline src/components/User/KycRequests/KycRequests.Queue.vue]: what confirmation?

File: src/components/User/KycRequests/KycRequests.Queue.vue [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        this.isDecisionEdited = false
```
Comment (max-nes) [inline src/components/User/KycRequests/KycRequests.Queue.vue]: eidtted and what?

File: src/components/User/KycRequests/KycRequests.Queue.vue [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+          <div
+            v-if="isCurrentDecisionShown"
+            class="kyc-requests-queue__current-decision"
+          >
+            <h3>Current decision</h3>
+            <review-decision-viewer :decision="currentDecision" />
+          </div>
+
+          <pending-request-actions
```
Comment (max-nes) [inline src/components/User/KycRequests/KycRequests.Queue.vue]: queue and decision

File: src/components/User/KycRequests/components/PendingRequestActions.vue [hunk 6]
```diff
@@ -0,0 +1,160 @@
+<template>
```
Comment (max-nes) [inline src/components/User/KycRequests/components/PendingRequestActions.vue]: actually works with decisions of the queue but the path of the file is confusing

File: src/components/User/KycRequests/components/ReviewSummary.vue [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            @click="emitDecisionEditing"
+          />
+        </div>
+      </template>
+
+      <template v-else>
+        <p class="app__block">
+          You haven't reviewed any request
+        </p>
+      </template>
```
Comment (max-nes) [inline src/components/User/KycRequests/components/ReviewSummary.vue]: remove

File: src/components/User/KycRequests/components/ReviewSummary.vue [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+        decision.setReviewedState()
+      } catch (e) {
+        const errorMessage = ErrorHandler.extractErrorMessage(e)
+        decision.setErrorState(errorMessage)
+
+        ErrorHandler.processWithoutFeedback(e)
+      }
+    },
+
+    createDecisionOperation (decision) {
```
Comment (max-nes) [inline src/components/User/KycRequests/components/ReviewSummary.vue]: createReviewOperation or decisionToReviewOperation

File: src/components/User/KycRequests/wrappers/ReviewDecision.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  get isReadyForReview () {
+    const readyForReviewStates = [
+      DECISION_STATES.approve,
+      DECISION_STATES.reject,
+    ]
+
+    return readyForReviewStates.includes(this.state)
+  }
+
+  setReview ({ action, reason = '' }) {
```
Comment (max-nes) [inline src/components/User/KycRequests/wrappers/ReviewDecision.js]: approve, skip, reject(msg)

File: src/components/User/KycRequests/wrappers/ReviewDecision.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    this.action = action
+    this.state = DECISION_STATES[this.action]
+
+    this.reason = reason
+    this.errorMessage = ''
+  }
+
+  setProcessingState () {
```
Comment (max-nes) [inline src/components/User/KycRequests/wrappers/ReviewDecision.js]: state is redundant

Comment (ardier16) [reply src/components/User/KycRequests/wrappers/ReviewDecision.js]: Removed

File: src/components/User/KycRequests/wrappers/ReviewDecision.js (hunk 9 above)
Comment (ardier16) [reply src/components/User/KycRequests/wrappers/ReviewDecision.js]: Updated

File: src/components/User/KycRequests/components/ReviewSummary.vue (hunk 8 above)
Comment (ardier16) [reply src/components/User/KycRequests/components/ReviewSummary.vue]: Renamed

File: src/components/User/KycRequests/components/ReviewSummary.vue (hunk 7 above)
Comment (ardier16) [reply src/components/User/KycRequests/components/ReviewSummary.vue]: Removed

File: src/components/User/KycRequests/components/PendingRequestActions.vue (hunk 6 above)
Comment (ardier16) [reply src/components/User/KycRequests/components/PendingRequestActions.vue]: Extracted to `/queue` folder

File: src/components/User/KycRequests/KycRequests.Queue.vue (hunk 5 above)
Comment (ardier16) [reply src/components/User/KycRequests/KycRequests.Queue.vue]: Renamed components

File: src/components/User/KycRequests/KycRequests.Queue.vue (hunk 4 above)
Comment (ardier16) [reply src/components/User/KycRequests/KycRequests.Queue.vue]: Renamed

File: src/components/User/KycRequests/KycRequests.Queue.vue (hunk 3 above)
Comment (ardier16) [reply src/components/User/KycRequests/KycRequests.Queue.vue]: Renamed

File: src/components/User/KycRequests/KycRequests.Queue.vue (hunk 2 above)
Comment (ardier16) [reply src/components/User/KycRequests/KycRequests.Queue.vue]: Removed

File: src/components/User/KycRequests/KycRequests.Queue.vue (hunk 1 above)
Comment (ardier16) [reply src/components/User/KycRequests/KycRequests.Queue.vue]: Shown

**Pull request merged**